src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -4764,12 +4765,47 @@ private void AddSynthesizedTupleMembersIfNecessary(MembersAndInitializersBuilder
                         }
                         break;
 
+                    case SyntaxKind.IncompleteMember:
+                        {
+                            var incompleteMemberSyntax = (IncompleteMemberSyntax)m;
+                            if (incompleteMemberSyntax.Type is not null)
+                            {
+                                // We want to produce binding errors for incomplete members
+                                _ = bodyBinder.BindType(incompleteMemberSyntax.Type, diagnostics);
+                            }
+
+                            // Breaks a possible cycle when attribute arguments binding wants to get members.
+                            if (_isBindingAttribute)
+                            {
+                                break;
+                            }
+
+                            var incompleteMemberDiagnostics = BindingDiagnosticBag.GetInstance();
+                            foreach (var attributeList in incompleteMemberSyntax.AttributeLists)
+                            {
+                                foreach (var attribute in attributeList.Attributes)
+                                {
+                                    var attributeType = bodyBinder.BindType(attribute.Name, incompleteMemberDiagnostics).Type;
+                                    Debug.Assert(attributeType is NamedTypeSymbol);
+                                    if (attributeType is NamedTypeSymbol attributeNamedType)
+                                    {
+                                        _isBindingAttribute = true;
+                                        bodyBinder.GetAttribute(attribute, attributeNamedType, beforeAttributePartBound: null, afterAttributePartBound: null, incompleteMemberDiagnostics);
+                                        _isBindingAttribute = false;
```

Comment:
Probably not a thread-safe approach?

I also dislike this idea as a whole anyway 😕

Reply:
Should I introduce a new symbol, add it to `builder.NonTypeMembers`, and let its `ForceComplete` bind the attributes? Similar to what event symbols do?

https://github.com/dotnet/roslyn/blob/87bf0d0026d379f9434a2d6ee758aa25ed327ae2/src/Compilers/CSharp/Portable/Symbols/Source/SourceEventSymbol.cs#L63-L76

https://github.com/dotnet/roslyn/blob/87bf0d0026d379f9434a2d6ee758aa25ed327ae2/src/Compilers/CSharp/Portable/Symbols/Source/SourceEventSymbol.cs#L154-L208

What should the new symbol type derive from? `ErrorTypeSymbol` feels the most appropriate, but it's incorrect since we'll be adding the symbol to `NonTypeMembers`